keras_nlp/samplers/beam_sampler.py

```diff
@@ -36,11 +36,14 @@ class BeamSampler(Sampler):
     Args:
         num_beams: int. The number of beams that should be kept at each
             time-step. `num_beams` should be strictly positive.
+        return_all_beams: bool. When set to `True`, the sampler will return the top prompt,
+            all prompts and their respective probabilities score.
 
     Call Args:
         {{call_args}}
 
     Examples:
+    Example 1:
```

Comment:
Instead of Example 1,2,3... let's document what each examples does. For example, you can use "Only return the beam of largest accumulated probability" here, and "Return all beams and their probability" in the next example.